### PR TITLE
Add periodic table data and color scheme information

### LINKED ISSUE #223: Periodic table data
Pymatgen offers an extensive [periodic table yaml file](https://github.com/materialsproject/pymatgen/blob/7dd059cff3e1ed602565c7387e21e7dfa3e376b9/dev_scripts/periodic_table.yaml). We could, in principle, use that file to replace and extend the information in `src/molara/Molecule/atom.py`. As long as we credit the them, we are allowed to copy it (according to [this thread](https://github.com/materialsproject/pymatgen/issues/3579)). Does anyone have any objections?

### PR DESCRIPTION
**Pull request by ab5424**
Closes #223

## Summary by CodeRabbit

- **New Features**
	- Enhanced atomic number conversion to support hydrogen isotopes.
	- Implemented fetching and storage of periodic table data.
	- Added functionality to retrieve atom colors from external sources.

- **Documentation**
	- Included licensing details for the periodic table data.

- **Tests**
	- Updated test cases to accommodate new hydrogen isotope features.

- **Chores**
	- Configured pre-commit to exclude JSON files from YAML checks.

### CODE REVIEW

**Comment by ab5424** [comment]
Ready to merge. Next week, I will work on #211 so the user can select the color scheme in the new structure costumizer.

**Comment by ab5424** [comment]
Btw, I deliberately didn't add the libraries used in the developer scripts to the requirements/pyproject. Realistically speaking, these scripts will be executed less than once a year.

File: develop/periodic_table/get_pd_data.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+if __name__ == "__main__":
+    pt_data = fetch_periodic_table()
+    with open(file_path + "periodic_table.json", mode="w") as file:
+        json.dump(pt_data, file)
+
+    license_text = fetch_license()
+    with open(file_path + "periodic_table_copyright", mode="w") as file:
+        file.write("Periodic Table data from pymatgen.\nThe data is licensed under the following terms:\n\n")
+        file.write(license_text)
```
[the remark this answers is not in the document]
Comment (ab5424) [reply develop/periodic_table/get_pd_data.py:39]: Can you make a suggestion?

**Comment by Michel-Heinz** [comment]
Works just fine for me and now we have a good source for the information!
Does the MIT clash with our License?

**Comment by ab5424** [comment]
I don't think so, MIT should be more liberal and I made sure that the licenses text is included in the releases

See also https://opensource.stackexchange.com/questions/13029/how-mixing-mit-licensed-code-with-gplv3-project-works

**Pull request merged**